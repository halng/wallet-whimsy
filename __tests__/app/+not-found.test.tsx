/*
 * Copyright 2025 Hal Ng
 * Licensed under the Apache License, Version 2.0 (the License);
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an AS IS BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { render, screen } from '@testing-library/react-native';

import NotFoundScreen from '@/app/+not-found';

jest.mock('expo-router', () => ({
  Stack: {
    Screen: jest.fn(({ children }) => children),
  },
  Link: jest.fn(({ children }) => children),
}));
describe('App - NotFoundScreen', () => {
  test('Snapshot test ', async () => {
    render(<NotFoundScreen />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
  test('UI should render correctly', () => {
    render(<NotFoundScreen />);
    const msg = screen.getByText("This screen doesn't exist.");
    expect(msg).toBeTruthy();
  });
});
