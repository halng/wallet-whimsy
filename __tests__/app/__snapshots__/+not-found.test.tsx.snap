// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - NotFoundScreen Snapshot test  1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        {
          "fontSize": 32,
          "fontWeight": "bold",
          "lineHeight": 32,
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    This screen doesn't exist.
  </Text>
</View>
`;
